src/libraries/JANA/JApplication.h

```diff
@@ -173,6 +176,11 @@ JParameter* JApplication::SetDefaultParameter(std::string name, T& val, std::str
     return m_params->SetDefaultParameter(name.c_str(), val, description);
 }
 
+template <typename T>
+T JApplication::RegisterParameter(std::string name, const T default_val, std::string description) {
+    return m_params->RegisterParameter(name.c_str(), default_val, description);
```

Comment:
```suggestion
    return m_params->RegisterParameter(name, default_val, description);
```

Reply:
I actually thought about doing this too, but it would mean changing several other places where the same pattern exists. Without knowing the original motivation Nathan had, I decided to keep the changes minimal and consistent with the rest of the code.

Reply:
I just did a deep dive into the git blame. I did make the change, but I simply don't remember why. The context was as follows: I was doing the BDX port and discovering lots of problems that needed to be urgently fixed before anything would run at all. (The code was very rough then!) There definitely wasn't a deep, thoughtful reason behind it. Either we can leave it as a c_str just to be extra careful about not breaking things, or we could pass it as a std::string (which looks cleaner) and even std::move it for an infinitesimal performance improvement.